### PR TITLE
Allow decryption with revoked keys

### PR DESCRIPTION
**Pull request by larabr**
Ignore revocation status when filtering keys for decryption.

See #1126

### CODE REVIEW

File: test/general/openpgp.js [hunk 1]
```diff
@@ -2428,6 +2428,41 @@ describe('OpenPGP.js public api tests', function() {
...
+        const encrypted = await openpgp.encrypt({
+          message: openpgp.message.fromText(plaintext),
+          publicKeys: pubKeyDE
+        });
+        privKeyDE.subKeys[0] = await privKeyDE.subKeys[0].revoke(privKeyDE.primaryKey);
+        const decOpt = {
+          message: await openpgp.message.readArmored(encrypted.data),
+          privateKeys: privKeyDE
+        };
+        await expect(openpgp.decrypt(decOpt)).to.not.be.rejected;
```
Comment (twiss) [inline test/general/openpgp.js:2444]: This is a bit redundant with just calling the function. If you want to have an `expect` in there, I would write something like:
```suggestion
        const decrypted = await openpgp.decrypt(decOpt);
        expect(decrypted.data).to.equal(plaintext);
```

File: src/packet/public_key_encrypted_session_key.js [hunk 2]
```diff
@@ -128,6 +128,12 @@ PublicKeyEncryptedSessionKey.prototype.encrypt = async function (key) {
  */
 PublicKeyEncryptedSessionKey.prototype.decrypt = async function (key) {
   const algo = enums.write(enums.publicKey, this.publicKeyAlgorithm);
+  const expectedAlgo = enums.write(enums.publicKey, key.algorithm);
+  // check that session key algo matches the secret key algo
+  // considering rsa_encrypt and rsa_encrypt_sign as equal
+  if ((algo > 2 || expectedAlgo > 2) && algo !== expectedAlgo) {
```
Comment (twiss) [inline src/packet/public_key_encrypted_session_key.js:134]: I don't think you have to consider rsa_encrypt and rsa_encrypt_sign as equal. ProtonMail/crypto has this: https://github.com/ProtonMail/crypto/blob/1fa7f403fb9c3aa7b47443e1e3b39c80d6eedcb6/openpgp/packet/encrypted_key.go#L91-L93 and it hasn't caused any issues that I know of.

Comment (larabr) [reply src/packet/public_key_encrypted_session_key.js:134]: We have two tests that fail without this check - I think older keys might give issues in this sense

Comment (twiss) [reply src/packet/public_key_encrypted_session_key.js:134]: Maybe the tests should be fixed instead?
https://github.com/openpgpjs/openpgpjs/search?q=enc.publicKeyAlgorithm+%3D+%27rsa_encrypt%27

**Pull request merged**